microscopy_segmentation/unet_train_monai.py

```diff
@@ -0,0 +1,242 @@
+""" UNet training script for 2D microscopy SEM dataset
+Mostly based on https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py
+but adapted to create patches from the input images and have a multi-class model output.
+This script tries to reproduce most accurately this ivadomed config used in the default SEM model
+for ADS: https://github.com/axondeepseg/default-SEM-model/blob/main/model_seg_rat_axon-myelin_sem/model_seg_rat_axon-myelin_sem.json
+"""
+
+import logging
+import json
+import sys
+import tempfile
+from glob import glob
+from pathlib import Path
+
+import torch
+from PIL import Image
+from torch.utils.tensorboard import SummaryWriter
+import numpy as np
+
+import monai
+from monai.data import create_test_image_2d, pad_list_data_collate, list_data_collate, decollate_batch, DataLoader
+from monai.inferers import sliding_window_inference
+from monai.metrics import DiceMetric
+from monai.transforms import (
+    Activations,
+    EnsureChannelFirstd,
+    AsDiscrete,
+    Compose,
+    LoadImaged,
+    RandAffined,
+    Rand2DElasticd,
+    NormalizeIntensityd,
+    ResizeWithPadOrCropd
+)
+from monai.visualize import plot_2d_or_3d_image
+
+
+def main(working_dir):
```

Comment:
why making it a function (as opposed to a script)?

Reply:
@jcohenadad agreed. In this context, there should not be any command-line parsing.